sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_common/math.py

```diff
@@ -3,20 +3,44 @@
 # ---------------------------------------------------------
 
 import math
-from typing import List
+from typing import List, Callable, Any
 
 from azure.ai.evaluation._exceptions import EvaluationException, ErrorBlame, ErrorCategory, ErrorTarget
 
 
 def list_sum(lst: List[float]) -> float:
+    """Given a list of floats, return the sum of the values.
+    
+    :param lst: A list of floats.
+    :type lst: List[float]
+    :return: The sum of the values in the list.
+    :rtype: float
+    """
+
     return sum(lst)
 
 
 def list_mean(lst: List[float]) -> float:
+    """Given a list of floats, calculate the mean of the values.
+    
+    :param lst: A list of floats.
+    :type lst: List[float]
+    :return: The mean of the values in the list.
+    :rtype: float
+    """
+
     return list_sum(lst) / len(lst)
 
 
 def list_mean_nan_safe(lst: List[float]) -> float:
+    """Given a list of floats, remove all nan or None values, then calculate the mean of the remaining values.
+    
+    :param lst: A list of floats.
+    :type lst: List[float]
+    :return: The mean of the values in the list.
+    :rtype: float
+    """
+
     msg = "All score values are NaN. The mean cannot be calculated."
     if all(math.isnan(l) for l in lst):
         raise EvaluationException(
```

Comment:
Will evaluation fail if all values are NaN for an evaluator ?

Reply:
Yes. I think that's reasonable, since at that point there's no data to both aggregating, and the user should probably be aware that one of their evaluators failed to produce anything.

Reply:
Actually I'm going to catch and wrap this in a more informative error message, that way the user knows which column failed.

Reply:
This step comes after all evaluators have run if one fails (with all NaNs) other might still have valid values so should we fail in that case ?

Reply:
Changed to print a warning instead.

Reply:
Log* a warning.